src/libcore/pin.rs

```diff
@@ -7,23 +7,28 @@
 //! since moving an object with pointers to itself will invalidate them,
 //! which could cause undefined behavior.
 //!
-//! In order to prevent objects from moving, they must be pinned
-//! by wrapping a pointer to the data in the [`Pin`] type. A pointer wrapped
-//! in a `Pin` is otherwise equivalent to its normal version, e.g. `Pin<Box<T>>`
-//! and `Box<T>` work the same way except that the first is pinning the value
-//! of `T` in place.
+//! By default, all types in `Rust` are movable. Rust allows passing all types by-value,
+//! and common smart-pointer types such as `Box`, `Rc`, and `&mut` allow replacing and
+//! moving the values they contain. The [`Pin`] type provides a wrapper for these pointer
+//! types which prohibits moving the value behind the pointer. For example, `Pin<Box<T>>`
```

Comment:
Should we also mention here that `Pin` not only means you can't move it now, but it's also a "promise" that the `T` won't be moved in the future either.